.github/workflows/graphql-functional.yaml

```diff
@@ -0,0 +1,204 @@
+        description: "The folder where Magento will be installed"
+
+      magento_repository:
+        type: string
+        required: false
+        default: "https://mirror.mage-os.org/"
+        description: "Where to install Magento from"
+
+      matrix:
+        type: string
+        required: true
+        description: "The matrix of Magento versions to test against"
+
+      fail-fast:
+        type: boolean
+        required: false
+        default: true
+
+      test_command:
+        type: string
+        required: false
+        default: ../../../vendor/bin/phpunit
+        description: "The integration test command to run"
+
+      composer_cache_key:
+        type: string
+        required: false
+        default: ""
+        description: A key to version the composer cache. Can be incremented if you need to bust the cache.
+
+    secrets:
+      composer_auth:
+        required: false
+
+jobs:
+  api-functional-test:
+    name: ${{ matrix.magento }} GraphQL Test
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: ${{ inputs.fail-fast }}
+      matrix: ${{ fromJSON(inputs.matrix) }}
+
+    services:
+      elasticsearch:
+        image: ${{ matrix.elasticsearch }}
+        env:
+          # By default, ElasticSearch refuses to spawn in single node configuration, as it expects redundancy.
+          # This is a dev environment, so redundancy is just wasteful.
+          discovery.type: single-node
+          # Disable HTTPS and password authentication
+          # this is a local dev environment, so the added CA chain complexity is an extreme overkill
+          xpack.security.enabled: false
+          xpack.security.http.ssl.enabled: false
+          xpack.security.transport.ssl.enabled: false
+        options: >-
+          --health-cmd "curl http://localhost:9200/_cluster/health"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 10
+        ports:
+          - 9200:9200
+
+      mysql:
+        image: ${{ matrix.mysql }}
+        env:
+          MYSQL_DATABASE: magento_functional_tests
+          MYSQL_USER: user
+          MYSQL_PASSWORD: password
+          MYSQL_ROOT_PASSWORD: rootpassword
+        ports:
+          - 3306:3306
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+
+      rabbitmq:
+        image: ${{ matrix.rabbitmq }}
+        env:
+          RABBITMQ_DEFAULT_USER: guest
+          RABBITMQ_DEFAULT_PASS: guest
+        ports:
+          - 5672:5672
+          - 15672:15672
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set PHP Version
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          tools: composer:v${{ matrix.composer }}
+          coverage: none
+
+      - run: composer create-project --repository-url="${{ inputs.magento_repository }}" "${{ matrix.magento }}" ${{ inputs.magento_directory }} --no-install
```

Comment:
This should be reworked to use the new `setup-magento` and `cache-magento` commands.